Yesod/Auth/OAuth2/Google.hs

```diff
@@ -0,0 +1,118 @@
+                                  -> Text -- ^ Client ID
+                                  -> Text -- ^ Client secret
+                                  -> AuthPlugin m
+oauth2GoogleScopedWithCustomId toCreds scopes clientId clientSecret = authOAuth2 "google" oauth $ fetchGoogleProfile toCreds
+  where
+    oauth = OAuth2
+        { oauthClientId = encodeUtf8 clientId
+        , oauthClientSecret = encodeUtf8 clientSecret
+        , oauthOAuthorizeEndpoint = encodeUtf8 $
+                                    "https://accounts.google.com/o/oauth2/auth?scope=" <> T.intercalate "+" scopes
+        , oauthAccessTokenEndpoint = "https://www.googleapis.com/oauth2/v3/token"
+        , oauthCallback = Nothing
+        }
+
+
+
+fetchGoogleProfile :: (GoogleUser -> AccessToken -> Creds m) -> Manager -> AccessToken -> IO (Creds m)
+fetchGoogleProfile toCreds manager token = do
+  userInfo <- authGetJSON manager token "https://www.googleapis.com/oauth2/v3/userinfo"
+  case userInfo of
+   Right user -> return $ toCreds user token
+   Left err -> throwIO $ InvalidProfileResponse "google" err
+
+data GoogleUser = GoogleUser
+                  { googleUserId :: Text
+                  , googleUserName :: Text
+                  , googleUserEmail :: Text
+                  , googleUserPicture :: Text
+                  , googleUserGivenName :: Text
+                  , googleUserFamilyName :: Text
+                  , googleUserHostedDomain :: Maybe Text
+                  }
+
+instance FromJSON GoogleUser where
+  parseJSON (Object o) = GoogleUser
+                         <$> o .: "sub"
+                         <*> o .: "name"
+                         <*> o .: "email"
+                         <*> o .: "picture"
+                         <*> o .: "given_name"
+                         <*> o .: "family_name"
+                         <*> o .:? "hd"
+
+  parseJSON _ = mzero
+  
+
+googleUid :: GoogleUser -> AccessToken -> Creds m
+googleUid = uidBuilder $ ("google-uid:" <>) . googleUserId
+
+
+emailUid :: GoogleUser -> AccessToken -> Creds m
+emailUid = uidBuilder googleUserEmail
+
+uidBuilder :: (GoogleUser -> Text) -> GoogleUser -> AccessToken -> Creds m
+uidBuilder f user token = Creds { credsPlugin = "google"
+                            , credsIdent = f user
+                            , credsExtra =
+                              [ ("email", googleUserEmail user)
+                              , ("name", googleUserName user)
+                              , ("given_name", googleUserGivenName user)
+                              , ("family_name", googleUserFamilyName user)
+                              , ("avatar_url", googleUserPicture user)
+                              , ("access_token", decodeUtf8 $ accessToken token)
+                              ] ++ maybeHostedDomain
+                            }
+  where maybeHostedDomain = maybeToList $ ((,) "hosted_domain") `fmap` googleUserHostedDomain user
```

Comment:
Can you line-break before `maybeHostedDomain` and indent it 2 spaces beyone the `where`?

Can you use `(<$>)` in place of infix `fmap`?